### PR TITLE
fix: Improve form inputs

### PR DESCRIPTION
**Pull request by mahatoankitkumar**
## Problem
Better form fields
- enum → Toggle
- boolean → dropdown with true and false options

https://github.com/user-attachments/assets/5152c2fb-316d-4617-86aa-09d90460ec8c

### CODE REVIEW

**Review by Datron — APPROVED** [review]
Lets get this in

**Review by pratikmishra356 — APPROVED** [review]

File: crates/frontend/src/components/default_config_form.rs [hunk 1]
```diff
@@ -203,52 +205,82 @@ where
...
+                    }
+                    "NUMBER" | "INTEGER" => {
+                        view! {
+                            <input
+                                type="number"
+                                placeholder="Value"
+                                class="input input-bordered w-full max-w-md"
+                                value=config_value.get()
+                                on:change=move |ev| {
+                                    logging::log!("{:?}", event_target_value(& ev));
```
Comment (sauraww) [inline crates/frontend/src/components/default_config_form.rs:250]: Can you add the desciption for the log here ?

File: crates/frontend/src/components/default_config_form.rs [hunk 2]
```diff
@@ -203,52 +205,82 @@ where
...
-                                </label>
-                                {input_format}
-                            </div>
-                            <div class="divider"></div>
+                            <textarea
+                                type="text"
+                                placeholder="Value"
+                                class="input input-bordered w-full max-w-md pt-3"
+                                on:change=move |ev| {
+                                    logging::log!("{:?}", event_target_value(& ev));
```
Comment (sauraww) [inline crates/frontend/src/components/default_config_form.rs:264]: Here as well

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/input_components.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+#[component]
+pub fn enum_dropdown(
+    schema: Map<String, Value>,
+    config_value: String,
+    handle_change: Callback<String, ()>,
+    #[prop(default = String::new())] class: String,
+    #[prop(default = false)] disabled: bool,
+) -> impl IntoView {
+    let (value, set_value) = create_signal(config_value.replace("\\", ""));
```
Comment (sauraww) [inline crates/frontend/src/components/input_components.rs:14]: Can we avoid replacing it ? 
What if someone actually wants to send this in the input ?

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/components/dimension_form.rs [hunk 4]
```diff
@@ -209,8 +209,9 @@ where
                                     ev.prevent_default();
                                 }
                             }
+
                             on:change=move |ev| {
-                                logging::log!("{:?}", event_target_value(&ev).parse::<u32>());
+                                logging::log!("{:?}", event_target_value(& ev).parse::< u32 > ());
```
Comment (sauraww) [inline crates/frontend/src/components/dimension_form.rs:214]: Can you add the log description here as well ?

**Review by sauraww — CHANGES_REQUESTED** [review]

File: crates/frontend/src/utils.rs [hunk 5]
```diff
@@ -516,3 +516,14 @@ pub fn set_local_storage(_key: &str, _value: &str) -> Option<()> {
         }
     }
 }
+
+pub fn get_key_type(schema: &Map<String, Value>) -> String {
+    if schema.contains_key("enum") {
+        String::from("ENUM")
+    } else {
+        match schema.get("type").unwrap_or(&Value::Null) {
```
Comment (sauraww) [inline crates/frontend/src/utils.rs:524]: Why are we using unwrap and match both here ? 
Possible to change it to : 
`match schema.get("type") {
    Some(Value::String(str_)) => str_.to_ascii_uppercase(),
    _ => String::from("STRING"),
}
`

**Review by sauraww — DISMISSED** [review]

**Pull request merged**